src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/Mocks/IRoslynServicesFactory.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Moq;
+using Microsoft.CodeAnalysis;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    internal static class IRoslynServicesFactory
+    {
+        public static Mock<IRoslynServices> Create()
```

Comment:
In general the pattern is that we have the factories return the actual service itself (i.e. mock.Object) and then you can use Mock.Get(object) to get back the mock (for implement or Verify). It's odd to have to know that the Factory for x doesn't generate x.